### ISSUE TITLE
Cannot run the model on Windows. 

### ISSUE THREAD

**Issue by phuynhh** [Layout-Parser/layout-parser#15]
Hi, Thank you very much for your brilliant work.
I've just passed the first step to install the package on my Windows 10.  However, I am currently stuck with the second step to run the model. Would you have any hints to solve this error? 
Please check out the attached files below if you have a moment. 
Thank you very much. 

![Screenshot (8)](https://user-images.githubusercontent.com/68187026/114157069-25aa0c00-991b-11eb-8a0d-5e2820dd4f97.png)
![Screenshot (9)_LI](https://user-images.githubusercontent.com/68187026/114157091-2cd11a00-991b-11eb-9111-01a3b31b8989.jpg)

**Comment by maxml7331** [comment]
Push! Have exact the same error. Help would be greatly appreciated

**Comment by lolipopshock** [comment]
Hmmm very interesting. Looking into it.

**Comment by maxml7331** [comment]
> Hmmm very interesting. Looking into it.

Hey loliposhock! Thanks for the great work and your help is much appreciated. Could you tell me on which system setup (python version, windows/linux etc pp.) the package is working for you? I am not able to get the lp.detrectron running (no matter which different version etc I install that were mentioned by other people here in the threads). Thanks a lot!

**Comment by CrisChir** [comment]
Detectron2 on windows is not supported:

1. Can I install detectron2 on Win10? see https://github.com/facebookresearch/detectron2/issues/9
We do not officially support windows, but given that maskrcnn-benchmark can run on win10, it probably won't take a lot of changes to run detectron2 on win10. 

More details on the link

2.  There is conansherry implemtentation:
https://github.com/conansherry/detectron2

3. https://ivanpp.cc/detectron2-walkthrough-windows/

4. There are some reports that in version 0.3 It can be simply installed it with conda. but user report issues see the thread at point 1.

**Comment by fcakyon** [comment]
Have you considered using [mmdetection](https://pypi.org/project/mmdet/) that relies on [mmcv](https://pypi.org/project/mmcv-full/) and [mmpycocotools](https://pypi.org/project/mmpycocotools/), instead of detectron2? 

All models in detectron2 + tens of sota pretrained models are present in [mmdetection model zoo](https://github.com/open-mmlab/mmdetection/blob/master/docs/model_zoo.md).

Moreover, they support both windows and linux. They can be directly installed by `pip install mmdet` in all platforms.

**Comment by lolipopshock** [comment]
> Have you considered using [mmdetection](https://pypi.org/project/mmdet/) that relies on [mmcv](https://pypi.org/project/mmcv-full/) and [mmpycocotools](https://pypi.org/project/mmpycocotools/), instead of detectron2?
> 
> All models in detectron2 + tens of sota pretrained models are present in [mmdetection model zoo](https://github.com/open-mmlab/mmdetection/blob/master/docs/model_zoo.md).
> 
> Moreover, they support both windows and linux. They can be directly installed by `pip install mmdet` in all platforms.

Yes! It's actually on the roadmap, see https://github.com/Layout-Parser/layout-parser/blob/master/.github/CONTRIBUTING.md . And MMOCR is more relevant: https://github.com/open-mmlab/mmocr .

**Comment by fcakyon** [comment]
> > Have you considered using [mmdetection](https://pypi.org/project/mmdet/) that relies on [mmcv](https://pypi.org/project/mmcv-full/) and [mmpycocotools](https://pypi.org/project/mmpycocotools/), instead of detectron2?
> > All models in detectron2 + tens of sota pretrained models are present in [mmdetection model zoo](https://github.com/open-mmlab/mmdetection/blob/master/docs/model_zoo.md).
> > Moreover, they support both windows and linux. They can be directly installed by `pip install mmdet` in all platforms.
> 
> Yes! It's actually on the roadmap, see https://github.com/Layout-Parser/layout-parser/blob/master/.github/CONTRIBUTING.md . And MMOCR is more relevant: https://github.com/open-mmlab/mmocr .

Good to hear that. Thanks for providing the roadmap!

**Comment by kenho211** [comment]
Have the same error, but the cache dir is iopath_cache instead of fvcore_cache
![Capture](https://user-images.githubusercontent.com/32618031/114656350-7c02ba80-9d20-11eb-8ce2-0ea0ad1d7539.PNG)

**Comment by simonschoe** [comment]
push, same here (Win10 user)

**Comment by peter-malik** [comment]
A fix for this is to clone and modify iopath, since its a Windows10 file naming error.
So clone iopath (this version worked for me)
`git clone https://github.com/facebookresearch/iopath --single-branch --branch v0.1.8`

Change iopath/iopath/common/file_io.py; class: HTTPURLHandler; method: _get_local_path line 753 (line number might change with newer versions)
- from this:
				`filename = path.split("/")[-1]](url)`	
- to this:
				`filename = parsed_url.path.split("/")[-1]`

Then call:
`pip install -e iopath`
And it will work. 

Also detectron2v0.4 didnt work. Installed the latest one via:
`python -m pip install "git+https://github.com/facebookresearch/detectron2.git"`

The detectron2 installation will overwrite your modified iopath version because of dependency conflict, so its best to install iopath last. But it should work ok.

**Comment by simonschoe** [comment]
@peter-malik thanks for the guidance! I made it work that way, albeit a somwhat dirty workaround I must damit...

**Comment by phuynhh** [comment]
Hi @peter-malik, thank you very much, your solution worked on my win10.

**Comment by jacobcroope** [comment]
Doesn't address the problem of running natively on Windows but a possible workaround for Win 10 users.  I have a working install using WSL Ubuntu 20.0 and Jupyter-Notebook sideloaded on Linux and accessed through the browser.

**Comment by chevaliermr** [comment]
I have the same issue and have been unable to get the model path to work. Is there a fix for this soon?

**Comment by erelin6613** [comment]
As I am forced to work on Windows I downloaded config files and weights, thus feeding `config_path` and `model_path` to DetectronLayoutModel :P
As for detectron, indeed, it is pain in the neck to make it work on  Windows. That worked for me:

```
conda install pywin32
pip install Cython
pip install git+https://github.com/philferriere/cocoapi.git#subdirectory=PythonAPI
pip install git+https://github.com/DGMaxime/detectron2-windows.git
```
Hope it will help somebody :)

**Comment by tariqph** [comment]
> As I am forced to work on Windows I downloaded config files and weights, thus feeding `config_path` and `model_path` to DetectronLayoutModel :P As for detectron, indeed, it is pain in the neck to make it work on Windows. That worked for me:
> 
> ```
> conda install pywin32
> pip install Cython
> pip install git+https://github.com/philferriere/cocoapi.git#subdirectory=PythonAPI
> pip install git+https://github.com/DGMaxime/detectron2-windows.git
> ```
> 
> Hope it will help somebody :)

Thanks so much. This worked for me and saved me hours.

**Comment by Sreevatsava1** [comment]
One workaround I found is, to change the WEIGHTS path inside the config.yaml file.

If the path is like 
`WEIGHTS: https://www.dropbox.com/s/8v4uqmz1at9v72a/model_final.pth?dl=1`

Change it to
`WEIGHTS: https://www.dropbox.com/s/8v4uqmz1at9v72a/model_final.pth`

By removing the ?dl=1 at the end of the path resolved the OS Error for me.

**Comment by MarouaneZ1** [comment]
<img width="724" alt="erreurDetectron" src="https://user-images.githubusercontent.com/98398821/192034063-b1171356-c09f-4b51-81f0-2455ad18df91.PNG">
<img width="677" alt="erreurDetectronn" src="https://user-images.githubusercontent.com/98398821/192034065-3cbe1d16-4c4b-47c6-88d3-e3e8235b1ec4.PNG">
i'm trying to use detectron2 model for extraction information but i got this problem, i tried many ways but i failed to solve it, i tried to use google collab but it throws me :"ImportError: you cannot import is_directory",
does anyone know how can i fix it ?

**Comment by MarouaneZ1** [comment]
> One workaround I found is, to change the WEIGHTS path inside the config.yaml file.
> 
> If the path is like `WEIGHTS: https://www.dropbox.com/s/8v4uqmz1at9v72a/model_final.pth?dl=1`
> 
> Change it to `WEIGHTS: https://www.dropbox.com/s/8v4uqmz1at9v72a/model_final.pth`
> 
> By removing the ?dl=1 at the end of the path resolved the OS Error for me.

can you plz send me the config.yaml file plz

**Comment by Dimonka2** [comment]
Got the same error, just on a different file:

>OSError: [Errno 22] Invalid argument: 'C:\\Users\\XXXXX/.torch/iopath_cache\\s/ukbw5s673633hsw\\publaynet-tf_efficientdet_d0.pth.tar?dl=1.lock'

is it still unresolved?

**Comment by stapelberg** [comment]
Installing the latest version of iopath solved it for me, as it [contains a fix to strip query parameters when downloading models](https://github.com/facebookresearch/iopath/commit/a68affb73ccfa14bdbaba858a616ebd47aae6de1):

```
pip install -U 'git+https://github.com/facebookresearch/iopath@e348b6797c40c9eb4c96bf75e9aaf1b248297548'
```

Hopefully PR https://github.com/facebookresearch/detectron2/pull/5427/files can be updated and merged to fix this permanently.
